scalalib/src/mill/scalalib/PublishModule.scala

```diff
@@ -124,35 +124,73 @@ trait PublishModule extends JavaModule { outer =>
    * @param localIvyRepo The local ivy repository.
    *                     If not defined, the default resolution is used (probably `$HOME/.ivy2/local`).
    */
-  def publishLocal(localIvyRepo: String = null): define.Command[Unit] = T.command {
-    publishLocalTask(T.task {
-      Option(localIvyRepo).map(os.Path(_, T.workspace))
-    })()
+  def publishLocal(
+      localIvyRepo: String = null,
+      sources: Boolean = true,
+      doc: Boolean = true,
+      transitive: Boolean = true
```

Comment:
I agree that `transitive` should default to `false`. I ran into issues with it and some `publishLocal` tasks aggregation via `T.traverse` (because of duplicated artifacts it seems). Although starting from a blank slate, a default to `true` would have been nicer IMO.

Reply:
About `compileModuleDeps`, I'm not sure. My aim with `--transitive` is to allow users to use something they published straightaway. So the standard dependencies need to be published too, but the compile-only ones aren't needed. Users can resolve the published module fine without publishing the compile-only dependencies.